app/controllers/licence_controller.rb

```diff
@@ -0,0 +1,168 @@
+require "slimmer/headers"
+
+class LicenceController < ApplicationController
+  before_filter :redirect_if_api_request
+  before_filter -> { set_expiry unless viewing_draft_content? }
+  before_filter :set_edition_for_viewing_draft_content
+  before_filter -> { setup_content_item_and_navigation_helpers("/" + params[:slug]) }
+
+  helper_method :postcode
+
+  INVALID_POSTCODE = 'invalidPostcodeFormat'.freeze
+  NO_LOCATION_ERROR = "validPostcodeNoLocation".freeze
+  NO_MATCHING_AUTHORITY = 'noLaMatch'.freeze
+  NO_MAPIT_MATCH = 'fullPostcodeNoMapitMatch'.freeze
+
+  # NOTE: This is a temporary fix to ensure that these licences get treated as
+  # 'county/unitary' tiered (as opposed to 'district/unitary'). The tier data
+  # for licences used to be stored in LocalService model records in the content
+  # api, but the entries for the licences below have since been removed. In
+  # future this tier data will be stored in the licensing application
+  # (Licensify). Once that has happened and Frontend has been updated to use
+  # that tier information, this list and related code in the
+  # `appropriate_slug_from_location` method can be removed
+  LICENCE_SLUGS_WITH_COUNTY_TIER_OVERRIDE = [
+    'scaffolding-and-hoarding-licence',
+    'skip-operator-licence',
+    'permission-to-place-tables-and-chairs-on-the-pavement',
+    'pavement-or-street-display-licence',
+    'petroleum-storage-licence',
+    'weighbridge-operator-certificate',
+    'performing-animals-registration',
+    'approval-of-premises-for-civil-marriage-or-civil-partnership',
+    'licence-projection-over-highway-england-wales',
+  ].freeze
+
+  def search
```

Comment:
Would it be possible to further highlight which code-paths are handling the `post` method? I assume they are almost entirely seperate